src/toDo.js

```diff
@@ -21,15 +21,21 @@ class ToDoList {
   displayTask(taskObj) {
     const taskList = document.getElementById('main-list');
     const task = document.createElement('LI');
-    if (taskObj.completed) {
-      task.innerHTML = `<label><input type="checkbox" checked id="cbox${taskObj.index}
-      ">${taskObj.description}</label>`;
-    } else {
-      task.innerHTML = `<label><input type="checkbox" id="cbox${taskObj.index}
-      ">${taskObj.description}</label>`;
-    }
+    const taskId = `cbox${taskObj.index}`;
+    task.innerHTML = `<label><input type="checkbox" ${taskObj.completed ? 'checked' : ''} 
+      id="${taskId}">${taskObj.description}</label>`;
     task.classList.add('task-item');
     taskList.appendChild(task);
+    const checkbox = document.getElementById(taskId);
+    const self = this;
+    checkbox.addEventListener('change', () => {
+      if (this.checked) {
+        taskObj.completed = true;
+      } else {
+        taskObj.completed = false;
+      }
+      self.updateTask(taskObj);
+    });
```

Comment:
- [ ] good job on creating the event listener 👍 however it is not working as expected, when selecting the box the status of `completed` doesn't change. this is because you check if `this` is checked, however, `this` is actually the object pulled from the local storage, so in order to see if your object complete status is `true` or `false` you need to go inside that object by using `this.data` and after that, you still have to select the right task based on the index, after that you are in the actual item at which point you can check the status of `completed`. in the end, it could look something like below:
```
checkbox.addEventListener('change', () => {
      const object = this.data[taskObj.index - 1]
      if (object.completed) {
```
you see in the object constant I created I'm going to the actual task and then in the if statement you can check what the status of `completed` is, keep in mind that if the status is `false` you should change it to `true` and the other way around, right now the if statement checks if the status if `true` and if it is it changes it to `true` which is not correct.

Reply:
First of all, thanks for your comprehensive review. However, that's was not the issue, the [problem](https://www.loom.com/share/87489e046f1e40e990e6a9803d6ac0e4) was in the line 31, instead of 
`checkbox.addEventListener('change', () => {`
it should be
`checkbox.addEventListener('change', function () {`

[Brief explanation](https://stackoverflow.com/questions/57799578/why-do-function-declarations-get-hoisted-and-function-expressions-dont)

Reply:
Hi John,
I noticed that as well and in this project it works fine 👏 However in future projects you will be expected to replace the `function` keyword with arrow functions in accordance with ES6 Syntax. I made this suggestion to avoid confusion with future projects.